File: 0485-max-consecutive-ones/solution.go
func findMaxConsecutiveOnes(nums []int) int {
    count := 0
    maxCount := 0
    for i := 0; i < len(nums); i++ {
        if nums[i] != 1 {
            if count > maxCount {
                maxCount = count
            }
            count = 0
        } else {
            count++
        }
    }

    // for last index
    if count > maxCount {
        maxCount = count
    }

    return maxCount
}

File: 0485-max-consecutive-ones/solution.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0
        max_count = 0
        for i in nums:
            if i != 1:
                count = 0
                continue

            count += 1
            if count > max_count:
                max_count = count
        return max_count

